vm/src/vm/vm_new.rs

```diff
@@ -607,8 +547,32 @@ impl VirtualMachine {
         )
     }
 
-    pub fn new_eof_error(&self, msg: String) -> PyBaseExceptionRef {
-        let eof_error = self.ctx.exceptions.eof_error.to_owned();
-        self.new_exception_msg(eof_error, msg)
-    }
+    define_exception_fn!(new_lookup_error, lookup_error);
+    define_exception_fn!(new_eof_error, eof_error);
+    define_exception_fn!(new_attribute_error, attribute_error);
+    define_exception_fn!(new_type_error, type_error);
+    define_exception_fn!(new_os_error, os_error);
+    define_exception_fn!(new_system_error, system_error);
+
+    // TODO: remove & replace with new_unicode_decode_error_real
+    define_exception_fn!(new_unicode_decode_error, unicode_decode_error);
+
+    // TODO: remove & replace with new_unicode_encode_error_real
+    define_exception_fn!(new_unicode_encode_error, unicode_encode_error);
+
+    define_exception_fn!(
+        new_value_error,
+        value_error,
+        doc = "Create a new python ValueError object.
```

Comment:
Thought of have this doc as part of the macro and only put the python exception name to template.
so it can look like:
```rust
define_exception_fn!(new_value_error, value_error, ValueError);
define_exception_fn!(new_index_error, index_error, IndexEror);
```

I can do it in a separate PR if you think that it would be beneficial